### ISSUE TITLE
Support python 3.11

### ISSUE THREAD

**Issue by natestemen** [unitaryfund/mitiq#1560]
It's supposed to arrive in 5 days ([schedule](https://peps.python.org/pep-0664/)). There are some serious speed improvements.

**Comment by nathanshammah** [comment]
+1 for supporting this, and indeed it's been released.

**Comment by natestemen** [comment]
Thanks Peter for letting us know! I've opened a [PR](https://github.com/aws/amazon-braket-sdk-python/pull/479) on `braket` to loosen the oqpy dependency requirement so we don't have to do this tango so often. Hopefully we can work something out. Once that (or something similar) is merged, we can move forward on this.

**Comment by karalekas** [comment]
I doubt they'll elect to fully unpin OQpy as it's such a critical dependency for the Braket SDK. There shouldn't need to be much of a tango after this though? Supporting a new Python version only happens once a year

**Comment by natestemen** [comment]
> I doubt they'll elect to fully unpin OQpy as it's such a critical dependency for the Braket SDK. 

Yeah, that makes sense. I'll see what they recommend. Maybe there's an acceptable range. If not, it's not a problem, but 2 weeks ago I bumped the oqpy version in braket ([PR](https://github.com/aws/amazon-braket-sdk-python/pull/469)) after the numpy and openpulse versions were loosed (in oqpy) as that was blocking us from upgrading to the latest version of braket.

This is the "tango" of dependency resolution that I referred to. You're right, hopefully this will be smoother going forward.
